src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs

```diff
@@ -21,6 +21,9 @@ namespace Microsoft.CodeAnalysis.Remote;
 internal sealed partial class AssetProvider(Checksum solutionChecksum, SolutionAssetCache assetCache, IAssetSource assetSource, ISerializerService serializerService)
     : AbstractAssetProvider
 {
+    private const int PooledChecksumArraySize = 256;
```

Comment:
let's live a little.  how about 1024 :)

Reply:
256 caught most of the items that I saw, after that they were sparse with values up to a couple thousand. I don't have a strong preference, but thought I would add that context. I'll gladly change if you prefer.

Separately, I just noticed that in lieu of the ReadOnlyMemory changes, I could have used ImmutableCollectionsMarshal.AsImmutableArray in SynchronizeAssetsAsync to get an ImmutableArray and kept passing that down as before. Do you prefer the use of ReadOnlyMemory or should I use ImmutableCollectionsMarshal.AsImmutableArray and change the supporting code back to ImmutableArray?

Reply:
Thought this through, and using ImmutableCollectionsMarshal.AsImmutableArray would mean that the size would need to be passed along, which is essentially just passing over a span. Will just keep the current approach unless I hear otherwise.